### PR TITLE
Use new antiforgery package

### PR DESCRIPTION
**Pull request by rynowak**
Renames: `AntiForgery` -> `IAntiforgery`

React to naming changes and cleanup in api. 

Simplify use of mocks

### CODE REVIEW

File: global.json [hunk 1]
```diff
@@ -1,3 +1,8 @@
 {
-    "projects": ["src", "test/WebSites", "samples"]
+    "projects": [ "src", "test/WebSites", "samples" ],
+    "sdk": {
+        "version": "1.0.0-beta6-12114",
+        "runtime": "clr",
+        "architecture": "x86"
+    }
```
Comment (rynowak) [inline global.json]: will revert

File: Mvc.sln [hunk 2]
```diff
@@ -174,6 +172,8 @@ Project("{8BB2217D-0F2D-49D1-97BC-3654ED321F3B}") = "Microsoft.AspNet.Mvc.Extens
 EndProject
 Project("{8BB2217D-0F2D-49D1-97BC-3654ED321F3B}") = "Microsoft.AspNet.Mvc.Extensions.Test", "test\Microsoft.AspNet.Mvc.Extensions.Test\Microsoft.AspNet.Mvc.Extensions.Test.xproj", "{5DF6EFA5-865E-450B-BF83-DE9CE88EB77C}"
 EndProject
+Project("{8BB2217D-0F2D-49D1-97BC-3654ED321F3B}") = "AntiforgeryTokenWebSite", "test\WebSites\AntiforgeryTokenWebSite\AntiforgeryTokenWebSite.xproj", "{A353B17E-A940-4CE8-8BF9-179E24A9041F}"
```
Comment (dougbu) [inline Mvc.sln]: You've demoted it, now just a token web site. Poor thing :smiley_cat:

File: src/Microsoft.AspNet.Mvc.Extensions/Filters/ValidateAntiforgeryTokenAttribute.cs [hunk 3]
```diff
@@ -2,19 +2,20 @@
...
 
 using System;
+using Microsoft.AspNet.Antiforgery;
 using Microsoft.Framework.DependencyInjection;
 
 namespace Microsoft.AspNet.Mvc
 {
     [AttributeUsage(AttributeTargets.Class | AttributeTargets.Method, AllowMultiple = false, Inherited = true)]
-    public class ValidateAntiForgeryTokenAttribute : Attribute, IFilterFactory, IOrderedFilter
+    public class ValidateAntiforgeryTokenAttribute : Attribute, IFilterFactory, IOrderedFilter
```
Comment (dougbu) [inline src/Microsoft.AspNet.Mvc.Extensions/Filters/ValidateAntiforgeryTokenAttribute.cs]: Note this is a compat break from MVC 5. Attribute is named `ValidateAntiForgeryTokenAttribute` there.

File: src/Microsoft.AspNet.Mvc.Extensions/Rendering/Html/DefaultHtmlGenerator.cs [hunk 4]
```diff
@@ -34,20 +35,20 @@ public class DefaultHtmlGenerator : IHtmlGenerator
         /// <summary>
         /// Initializes a new instance of the <see cref="DefaultHtmlGenerator"/> class.
         /// </summary>
-        /// <param name="antiForgery">The <see cref="AntiForgery"/> instance which is used to generate anti-forgery
+        /// <param name="antiforgery">The <see cref="Antiforgery"/> instance which is used to generate anti-forgery
```
Comment (dougbu) [inline src/Microsoft.AspNet.Mvc.Extensions/Rendering/Html/DefaultHtmlGenerator.cs]: Does this pass `xml-docs-test`? Should be `<see cref="IAntiforgery"/>`.

File: src/Microsoft.AspNet.Mvc.Extensions/Rendering/Html/HtmlHelper.cs [hunk 5]
```diff
@@ -219,15 +219,10 @@ public virtual void Contextualize([NotNull] ViewContext viewContext)
...
         {
-            var tagBuilder = _htmlGenerator.GenerateAntiForgery(ViewContext);
-            if (tagBuilder == null)
-            {
-                return HtmlString.Empty;
-            }
-
-            return tagBuilder.ToHtmlString(TagRenderMode.SelfClosing);
+            var html = _htmlGenerator.GenerateAntiforgery(ViewContext);
+            return html ?? HtmlString.Empty;
```
Comment (dougbu) [inline src/Microsoft.AspNet.Mvc.Extensions/Rendering/Html/HtmlHelper.cs]: Why not do the `HtmlString.Empty` thing in `DefaultHtmlGenerator`? No big deal if the tag helper gets `PostContent` a bit more often.

File: src/Microsoft.AspNet.Mvc.Extensions/Rendering/IHtmlHelper.cs [hunk 6]
```diff
@@ -103,7 +103,7 @@ public interface IHtmlHelper
         /// &lt;form&gt; is submitted.
         /// </summary>
         /// <returns>A new <see cref="HtmlString"/> containing the &lt;hidden&gt; element.</returns>
-        HtmlString AntiForgeryToken();
+        HtmlString AntiforgeryToken();
```
Comment (dougbu) [inline src/Microsoft.AspNet.Mvc.Extensions/Rendering/IHtmlHelper.cs]: Another compat break from MVC 5. Previous name matches that version.

File: src/Microsoft.AspNet.Mvc.Extensions/Rendering/Html/IHtmlGenerator.cs [hunk 7]
```diff
@@ -31,7 +31,7 @@ public interface IHtmlGenerator
             object routeValues,
             object htmlAttributes);
 
-        TagBuilder GenerateAntiForgery([NotNull] ViewContext viewContext);
+        HtmlString GenerateAntiforgery([NotNull] ViewContext viewContext);
```
Comment (dougbu) [inline src/Microsoft.AspNet.Mvc.Extensions/Rendering/Html/IHtmlGenerator.cs]: Please add docs for this method.

File: src/Microsoft.AspNet.Mvc.TagHelpers/FormTagHelper.cs [hunk 8]
```diff
@@ -21,7 +21,7 @@ namespace Microsoft.AspNet.Mvc.TagHelpers
     public class FormTagHelper : TagHelper
     {
         private const string ActionAttributeName = "asp-action";
-        private const string AntiForgeryAttributeName = "asp-anti-forgery";
+        private const string AntiforgeryAttributeName = "asp-anti-forgery";
```
Comment (dougbu) [inline src/Microsoft.AspNet.Mvc.TagHelpers/FormTagHelper.cs]: Since you're such an "anti" hater, why didn't you rename this attribute `asp-antiforgery`. Looks inconsistent as-is.

File: test/Microsoft.AspNet.Mvc.Extensions.Test/Rendering/DefaultTemplatesUtilities.cs [hunk 9]
```diff
@@ -236,7 +237,7 @@ public static HtmlHelper<TModel> GetHtmlHelper<TModel>(TModel model, IModelMetad
             if (htmlGenerator == null)
             {
                 htmlGenerator = new DefaultHtmlGenerator(
-                    GetAntiForgeryInstance(),
+                    GetAntiforgeryInstance(),
```
Comment (dougbu) [inline test/Microsoft.AspNet.Mvc.Extensions.Test/Rendering/DefaultTemplatesUtilities.cs]: Remove this method and just inline `Mock.Of<IAntiforgery>()`

File: test/Microsoft.AspNet.Mvc.FunctionalTests/AntiforgeryTests.cs [hunk 10]
```diff
@@ -50,12 +50,12 @@ public class AntiForgeryTests
 
             // do a get response.
             var getResponse = await client.GetAsync("http://localhost/Account/Login");
-            var resposneBody = await getResponse.Content.ReadAsStringAsync();
+            var responseBody = await getResponse.Content.ReadAsStringAsync();
```
Comment (dougbu) [inline test/Microsoft.AspNet.Mvc.FunctionalTests/AntiforgeryTests.cs]: Nice. At least the class was previously consistent :smiling_imp:

File: test/Microsoft.AspNet.Mvc.TagHelpers.Test/TestableHtmlGenerator.cs [hunk 11]
```diff
@@ -38,7 +39,7 @@ public TestableHtmlGenerator(IModelMetadataProvider metadataProvider, IUrlHelper
             IOptions<MvcViewOptions> options,
             IUrlHelper urlHelper,
             IDictionary<string, object> validationAttributes)
-            : base(GetAntiForgery(), options, metadataProvider, urlHelper, new CommonTestEncoder())
+            : base(GetAntiforgery(), options, metadataProvider, urlHelper, new CommonTestEncoder())
```
Comment (dougbu) [inline test/Microsoft.AspNet.Mvc.TagHelpers.Test/TestableHtmlGenerator.cs]: Remove `GetAntiforgery()` method and just inline `Mock.Of<IAntiforgery>()`.

File: test/WebSites/AntiforgeryTokenWebSite/AntiforgeryTokenWebSite.xproj [hunk 12]
```diff
@@ -10,6 +10,9 @@
     <BaseIntermediateOutputPath Condition="'$(BaseIntermediateOutputPath)'=='' ">..\..\..\artifacts\obj\$(MSBuildProjectName)</BaseIntermediateOutputPath>
     <OutputPath Condition="'$(OutputPath)'=='' ">..\..\..\artifacts\bin\$(MSBuildProjectName)\</OutputPath>
   </PropertyGroup>
+  <PropertyGroup Label="Configuration">
+    <RootNamespace>AntiforgeryWebSite</RootNamespace>
+  </PropertyGroup>
```
Comment (dougbu) [inline test/WebSites/AntiforgeryTokenWebSite/AntiforgeryTokenWebSite.xproj]: Did VS add this noise? If not, revert.

File: test/WebSites/AntiforgeryTokenWebSite/readme.md [hunk 13]
```diff
@@ -0,0 +1,5 @@
+AntiforgeryTokenWebSite
+===
+
+This web site illustrates how to use antiforgery system to prevent CSRF attacks. The web site has an
```
Comment (dougbu) [inline test/WebSites/AntiforgeryTokenWebSite/readme.md]: minor: _the_ antiforgery system

File: test/WebSites/AntiforgeryTokenWebSite/readme.md [hunk 14]
```diff
@@ -0,0 +1,5 @@
...
+`AccountsController` which uses an anti forgery token to validate incoming form posts.
```
Comment (dougbu) [inline test/WebSites/AntiforgeryTokenWebSite/readme.md]: Why is "antiforgery" (hmm, nothing thinks that's a word) one word above and two here?

**Comment by dougbu** [comment]
:watch:

File: src/Microsoft.AspNet.Mvc.Extensions/Rendering/IHtmlHelper.cs (hunk 6 above)
Comment (rynowak) [inline src/Microsoft.AspNet.Mvc.Extensions/Rendering/IHtmlHelper.cs]: Yep. Will discuss with @Eilon and get a decision to either keep this or revert it.

File: src/Microsoft.AspNet.Mvc.Extensions/Rendering/Html/HtmlHelper.cs (hunk 5 above)
Comment (rynowak) [inline src/Microsoft.AspNet.Mvc.Extensions/Rendering/Html/HtmlHelper.cs]: I suspect this is a safeguard against someone implementing `IHtmlGenerator` and returning null.

`DefaultAntiforgery` and `DefaultHtmlGenerator` never return null or empty.

File: src/Microsoft.AspNet.Mvc.TagHelpers/FormTagHelper.cs (hunk 8 above)
Comment (rynowak) [inline src/Microsoft.AspNet.Mvc.TagHelpers/FormTagHelper.cs]: sure

File: test/WebSites/AntiforgeryTokenWebSite/AntiforgeryTokenWebSite.xproj (hunk 12 above)
Comment (rynowak) [inline test/WebSites/AntiforgeryTokenWebSite/AntiforgeryTokenWebSite.xproj]: VS did this, but I'll change it to the correct namespace.

File: test/WebSites/AntiforgeryTokenWebSite/readme.md (hunk 14 above)
Comment (rynowak) [inline test/WebSites/AntiforgeryTokenWebSite/readme.md]: because I missed that one :)

File: src/Microsoft.AspNet.Mvc.Extensions/Rendering/IHtmlHelper.cs (hunk 6 above)
Comment (rynowak) [inline src/Microsoft.AspNet.Mvc.Extensions/Rendering/IHtmlHelper.cs]: Will revert this change and the change to `[ValidateAntiForgery]`

File: src/Microsoft.AspNet.Mvc.Extensions/Filters/ValidateAntiforgeryTokenAttribute.cs (hunk 3 above)
Comment (rynowak) [inline src/Microsoft.AspNet.Mvc.Extensions/Filters/ValidateAntiforgeryTokenAttribute.cs]: Reverted as discussed

File: src/Microsoft.AspNet.Mvc.Extensions/Rendering/Html/DefaultHtmlGenerator.cs (hunk 4 above)
Comment (rynowak) [inline src/Microsoft.AspNet.Mvc.Extensions/Rendering/Html/DefaultHtmlGenerator.cs]: Hadn't run it yet. You can safely assume that I'll find and fix all these.

File: src/Microsoft.AspNet.Mvc.Extensions/Rendering/Html/IHtmlGenerator.cs (hunk 7 above)
Comment (rynowak) [inline src/Microsoft.AspNet.Mvc.Extensions/Rendering/Html/IHtmlGenerator.cs]: Added

File: src/Microsoft.AspNet.Mvc.TagHelpers/FormTagHelper.cs (hunk 8 above)
Comment (rynowak) [inline src/Microsoft.AspNet.Mvc.TagHelpers/FormTagHelper.cs]: Sure

File: test/Microsoft.AspNet.Mvc.Extensions.Test/Rendering/DefaultTemplatesUtilities.cs (hunk 9 above)
Comment (rynowak) [inline test/Microsoft.AspNet.Mvc.Extensions.Test/Rendering/DefaultTemplatesUtilities.cs]: Fixed

File: test/Microsoft.AspNet.Mvc.FunctionalTests/AntiforgeryTests.cs (hunk 10 above)
Comment (rynowak) [inline test/Microsoft.AspNet.Mvc.FunctionalTests/AntiforgeryTests.cs]: Fixed other occurrences.

File: test/Microsoft.AspNet.Mvc.TagHelpers.Test/TestableHtmlGenerator.cs (hunk 11 above)
Comment (rynowak) [inline test/Microsoft.AspNet.Mvc.TagHelpers.Test/TestableHtmlGenerator.cs]: sure

File: test/WebSites/AntiforgeryTokenWebSite/readme.md (hunk 13 above)
Comment (rynowak) [inline test/WebSites/AntiforgeryTokenWebSite/readme.md]: sure

File: global.json (hunk 1 above)
Comment (dougbu) [inline global.json]: Danger Will... You haven't reverted this yet.

File: src/Microsoft.AspNet.Mvc.Extensions/Rendering/Html/HtmlHelper.cs (hunk 5 above)
Comment (dougbu) [inline src/Microsoft.AspNet.Mvc.Extensions/Rendering/Html/HtmlHelper.cs]: :ok_hand:

**Comment by dougbu** [comment]
:shipit:
